src/google-maps/map-polyline/map-polyline.spec.ts

```diff
@@ -0,0 +1,145 @@
+    const polylineConstructorSpy = createPolylineConstructorSpy(polylineSpy).and.callThrough();
+
+    const fixture = TestBed.createComponent(TestApp);
+    fixture.componentInstance.options = polylineOptions;
+    fixture.componentInstance.path = path;
+    fixture.detectChanges();
+
+    expect(polylineConstructorSpy).toHaveBeenCalledWith(expectedOptions);
+  });
+
+  it('exposes methods that provide information about the Polyline', () => {
+    const polylineSpy = createPolylineSpy(polylineOptions);
+    createPolylineConstructorSpy(polylineSpy).and.callThrough();
+
+    const fixture = TestBed.createComponent(TestApp);
+    const polylineComponent = fixture.debugElement.query(By.directive(
+        MapPolyline))!.injector.get<MapPolyline>(MapPolyline);
+    fixture.detectChanges();
+
+    polylineSpy.getDraggable.and.returnValue(true);
+    expect(polylineComponent.getDraggable()).toBe(true);
+
+    polylineSpy.getEditable.and.returnValue(true);
+    expect(polylineComponent.getEditable()).toBe(true);
+
+    polylineComponent.getPath();
+    expect(polylineSpy.getPath).toHaveBeenCalled();
+
+    polylineSpy.getVisible.and.returnValue(true);
+    expect(polylineComponent.getVisible()).toBe(true);
+  });
+
+  it('initializes Polyline event handlers', () => {
+    const polylineSpy = createPolylineSpy(polylineOptions);
+    createPolylineConstructorSpy(polylineSpy).and.callThrough();
+
+    const fixture = TestBed.createComponent(TestApp);
+    fixture.detectChanges();
+
+    expect(polylineSpy.addListener).toHaveBeenCalledWith('click', jasmine.any(Function));
+    expect(polylineSpy.addListener).not.toHaveBeenCalledWith('dblclick', jasmine.any(Function));
+    expect(polylineSpy.addListener).not.toHaveBeenCalledWith('drag', jasmine.any(Function));
+    expect(polylineSpy.addListener).not.toHaveBeenCalledWith('dragend', jasmine.any(Function));
+    expect(polylineSpy.addListener).not.toHaveBeenCalledWith('dragstart', jasmine.any(Function));
+    expect(polylineSpy.addListener).not.toHaveBeenCalledWith('mousedown', jasmine.any(Function));
+    expect(polylineSpy.addListener).not.toHaveBeenCalledWith('mousemove', jasmine.any(Function));
+    expect(polylineSpy.addListener).not.toHaveBeenCalledWith('mouseout', jasmine.any(Function));
+    expect(polylineSpy.addListener).not.toHaveBeenCalledWith('mouseover', jasmine.any(Function));
+    expect(polylineSpy.addListener).not.toHaveBeenCalledWith('mouseup', jasmine.any(Function));
+    expect(polylineSpy.addListener).toHaveBeenCalledWith('rightclick', jasmine.any(Function));
+  });
+});
+
+@Component({
+  selector: 'test-app',
+  template: `<google-map>
+                <map-polyline [options]="options"
+                              [path]="path"
+                              (polylineClick)="handleClick()"
+                              (polylineRightclick)="handleRightclick()">
+                </map-polyline>
+            </google-map>`,
```

Comment:
Not a big deal, but it seems like this template is 4 space indented vs 2.

Reply:
I think this is something unusual with how spacing in multi line template literals are being interpreted. It is two spaces on my IDE, I'm not sure if there's a setting I need to to change on my IDE or what but I'll look into it.